flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -158,10 +158,15 @@ public ResultPartitionID getPartitionId() {
 		return partitionId;
 	}
 
-	int getPartitionIndex() {
+	public int getPartitionIndex() {
```

Comment:
unrelated change: package private is enough now.

Reply:
The intent of this change is to reflect that you can refer to partitions (and input channels) by index now - they are first class citizens of the respective objects.